tests/test_projectbuilder.py

```diff
@@ -336,3 +333,55 @@ def test_not_dir_outdir(mocker, tmp_dir, test_flit_path):
 
     with pytest.raises(build.BuildException):
         builder.build('sdist', out)
+
+
+@pytest.fixture(scope='session')
+def demo_pkg_inline(tmp_path_factory):
+    # builds a wheel without any dependencies and with a console script demo-pkg-inline
+    tmp_path = tmp_path_factory.mktemp('demo-pkg-inline')
+    builder = build.ProjectBuilder(srcdir=os.path.join(os.path.dirname(__file__), 'packages', 'inline'))
+    out = tmp_path / 'dist'
+    builder.build('wheel', str(out))
+    return next(out.iterdir())
+
+
+@pytest.mark.isolated
+def test_build_with_dep_on_console_script(tmp_path, demo_pkg_inline, capfd, mocker):
+    """
+    All command-line scripts provided by the build-required packages must be present in the build environment's PATH.
+    """
+    # we first install install demo pkg inline as build dependency (as this provides a console script we can check)
+    # to validate backend invocations contain the correct path we use an inline backend that will fail, but first
+    # provides the PATH information (and validates shutil.which is able to discover the executable - as PEP states)
+    toml = '[build-system]\nrequires = ["demo_pkg_inline"]\nbuild-backend = "build"\nbackend-path = ["."]\n'
+    code = dedent(
+        """
+        import os
+        import sys
+        print("BB " + os.environ["PATH"])
+        if sys.version_info[0] == 3:
+            import shutil
+            exe_at = shutil.which("demo-pkg-inline")
+        else:
+            paths = os.environ["PATH"].split(os.pathsep)
+            exe_at = os.path.join(paths[0], "demo-pkg-inline{}".format(".exe" if sys.platform == "win32" else ""))
+            if not os.path.exists(exe_at):
+                exe_at = None
+        print("BB " + exe_at)
+    """
```

Comment:
Tab here. Is black running here?

Reply:
Where do you see tab? Triple strings are not changed by pre-commit. Yes it's running, the CI passed.

Reply:
Sorry, I was asking to add a tab. The opening quotes are at a different level. Does black format it like this? Seems weird.

```python
    code = dedent(
        """
        import os
        import sys
        print("BB " + os.environ["PATH"])
        if sys.version_info[0] == 3:
            import shutil
            exe_at = shutil.which("demo-pkg-inline")
        else:
            paths = os.environ["PATH"].split(os.pathsep)
            exe_at = os.path.join(paths[0], "demo-pkg-inline{}".format(".exe" if sys.platform == "win32" else ""))
            if not os.path.exists(exe_at):
                exe_at = None
        print("BB " + exe_at)
    """
    )
```

Reply:
Black does not move triple quotes (that would change the string’s content and is not safe to do in general).